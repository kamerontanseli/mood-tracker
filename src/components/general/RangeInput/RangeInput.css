.RangeInput {
  width: 100%;
}

.RangeInput {
  -webkit-appearance: none; /* Hides the slider so that custom slider can be made */
  width: 100%; /* Specific width is required for Firefox. */
  background: transparent; /* Otherwise white in Chrome */
}

.RangeInput::-webkit-slider-thumb {
  -webkit-appearance: none;
}

.RangeInput:focus {
  outline: none;
}

.RangeInput::-webkit-slider-thumb {
  -webkit-appearance: none;
  border: 0;
  height: 36px;
  width: 36px;
  border-radius: 36px;
  background: #fff;
  border: 2px solid #3f51b5;
  cursor: pointer;
  margin-top: -14px;
  box-shadow: 0px 0px 4px #cacaca, 0px 0px 1px #c1c1c1;
}

.RangeInput::-webkit-slider-runnable-track {
  width: 100%;
  height: 8.4px;
  cursor: pointer;
  background: #eee;
  border-radius: 4px;
  border: 0;
}

.RangeInput:focus::-webkit-slider-runnable-track {
  background: #eee;
}
