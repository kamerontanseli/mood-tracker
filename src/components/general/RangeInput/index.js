import "./RangeInput.css";
import React from "react";

const RangeInput = props => <input type="range" className="RangeInput" {...props} />

export default RangeInput;
