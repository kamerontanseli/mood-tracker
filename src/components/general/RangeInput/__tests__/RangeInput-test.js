import React from 'react'
import renderer from 'react-test-renderer'
import RangeInput from '..'

describe('render', () => {
  it('should render', () => {
    const snapshot = renderer.create(
      <RangeInput />
    );
    expect(snapshot).toMatchSnapshot()
  })
})