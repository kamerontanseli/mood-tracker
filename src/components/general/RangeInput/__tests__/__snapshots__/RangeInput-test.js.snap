// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render should render 1`] = `
<input
  className="RangeInput"
  type="range"
/>
`;
